tasks/setup-debian.yml

```diff
@@ -1,26 +1,37 @@
 ---
-- name: Update APT package cache
-  apt:
-    update_cache: true
-    cache_valid_time: 3600
-  when: ansible_distribution == "Ubuntu"
+- name: Setup WireGuard preference
+  copy:
+    src: debian/etc/apt/preferences.d/limit-unstable
+    dest: /etc/apt/preferences.d/limit-unstable
+    owner: root
+    group: root
+    mode: 0644
   tags:
     - wg-install
 
-- name: Install required packages
-  package:
-    name: "{{ item }}"
+- name: Add WireGuard key
+  apt_key:
+    keyserver: "keyserver.ubuntu.com"
+    id: "8B48AD6246925553"
     state: present
-  with_items:
-    - software-properties-common
-    - linux-headers-{{ ansible_kernel }}
   tags:
     - wg-install
 
 - name: Add WireGuard repository
   apt_repository:
-    repo: "ppa:wireguard/wireguard"
+    repo: "deb http://deb.debian.org/debian/ unstable main"
     state: present
     update_cache: yes
   tags:
     - wg-install
+
+- name: Get architecture
+  shell: dpkg --print-architecture
+  register: dpkg_arch
+  changed_when: False
+
+- name: Install kernel headers to compile wireguard with DKMS
+  apt:
+    name:
+      - "linux-headers-{{ dpkg_arch.stdout }}"
```

Comment:
There is a funny thing happening here by the way. In the `ansible -m setup localhost` output there is `facter_os.architecture` which contains `amd64`.

In `ansible_architecture` this is normalized to 'x86_64' and not usable.

Reply:
I can find this variables:
```
    "ansible_architecture": "x86_64",
    "ansible_machine": "x86_64",
    "ansible_userspace_architecture": "x86_64",
```
But none with `amd64`. But at least it is consistent and AFAIK it's also the official declaration for a x86 64bit CPU.

Reply:
x86_64 is the consistent way of describing the architecture. On some machines I see the following - but definitely out of scope for this change. Possibly good for the dkms-headers role:
```js
# ansible -m setup localhost | sed -e 's/.* =>//g' | jq '.ansible_facts.facter_os'
{
  "architecture": "amd64",
  "distro": {
    "codename": "buster",
    "description": "Debian GNU/Linux buster/sid",
    "id": "Debian",
    "release": {
      "full": "testing",
      "major": "testing"
    }
  },
  "family": "Debian",
  "hardware": "x86_64",
  "name": "Debian",
  "release": {
    "full": "buster/sid",
    "major": "buster/sid"
  },
  "selinux": {
    "enabled": false
  }
}
```